google-cloud-spanner/src/main/java/com/google/cloud/spanner/InstanceAdminClient.java

```diff
@@ -19,14 +19,140 @@
+   *
+   * InstanceConfigInfo instanceConfigInfo =
+   *     InstanceConfig.newBuilder(InstanceConfigId.of(projectId, instanceConfigId), baseConfig)
+   *         .setDisplayName(instanceConfigId)
+   *         .addReadOnlyReplicas(readOnlyReplicas)
+   *         .build();
+   *
+   * final OperationFuture<InstanceConfig, CreateInstanceConfigMetadata> operation =
+   *     instanceAdminClient.createInstanceConfig(instanceConfigInfo);
+   *
+   * InstanceConfig instanceConfig = op.get(10, TimeUnit.MINUTES)
+   * }</pre>
+   *
+   * <!--SNIPPET instance_admin_client_create_instance_config-->
+   */
+  default OperationFuture<InstanceConfig, CreateInstanceConfigMetadata> createInstanceConfig(
+      InstanceConfigInfo instanceConfig, CreateAdminAPIOption... options) throws SpannerException {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  /**
+   * Updates a custom instance config. This can not be used to update a Google managed instance config. The returned {@code Operation} can be used to track the progress of
+   * updating the instance. If the named instance config does not exist, a SpannerException is
+   * thrown. The request must include at least one field to update.
+   *
+   * <p>Only user managed configurations can be updated.
+   *
+   * <p>Immediately after the request returns:
+   *
+   * <ul>
+   *   <li>The instance config's {@code reconciling} field is set to true.
+   * </ul>
+   *
+   * While the operation is pending:
+   *
+   * <ul>
+   *   <li>Cancelling the operation sets its metadata's cancel_time.
+   *   <li>The operation is guaranteed to succeed at undoing all changes, after which point it
+   *       terminates with a `CANCELLED` status.
+   *   <li>All other attempts to modify the instance config are rejected.
+   *   <li>Reading the instance config via the API continues to give the pre-request values.
+   * </ul>
+   *
+   * Upon completion of the returned operation:
+   *
+   * <ul>
+   *   <li>Creating instances using the instance configuration uses the new values.
+   *   <li>The instance config's new values are readable via the API.
+   *   <li>The instance config's {@code reconciling} field becomes false.
+   * </ul>
+   *
+   * <!--SNIPPET instance_admin_client_update_instance_config-->
+   *
+   * <pre>{@code
+   * String projectId = "my-project";
+   * String instanceConfigId = "custom-user-config";
+   * String displayName = "my-display-name";
+   *
+   * InstanceConfigInfo instanceConfigInfo =
+   *     InstanceConfig.newBuilder(InstanceConfigId.of(projectId, instanceConfigId))
+   *         .setDisplayName(displayName)
+   *         .build();
+   *
+   * // Only update display name.
+   * final OperationFuture<InstanceConfig, UpdateInstanceConfigMetadata> operation =
+   *     instanceAdminClient.updateInstanceConfig(
+   *         instanceConfigInfo, ImmutableList.of(InstanceConfigField.DISPLAY_NAME));
+   *
+   * InstanceConfig instanceConfig = operation.get(5, TimeUnit.MINUTES);
```

Comment:
nit: should we use the same timeout in minutes for all operations? This one uses 5 minutes, while above we use 10 minutes. Does it mean this one is expected to be faster? If not, I would argue to use the same timeout.

Reply:
Changing both the timeouts to 5 mins makes sense.

Reply:
Done.